src/vivarium/framework/results/context.py

```diff
@@ -181,13 +182,21 @@ def gather_results(
             else:
                 if stratifications is None:
                     for observation in observations:
-                        df = observation.results_gatherer(filtered_pop)
-                        yield df, observation.name, observation.results_updater
+                        if not observation.to_observe(event):
```

Comment:
I think my slight preference would be to put more of this in `observation` if possible, since we are only calling methods from `observation`. Then one could do e.g. 
```
for observation in observations:
    yield observation.get_this_tuple_of_results_if_to_observe(event, filtered_pop)
```
(name not serious ofc)
My reasoning here would be that we can avoid deeply nesting conditions and encapsulate things from observation more (e.g. maybe you don't need to_observe in the interface if you do this but need a different method instead. Doesn't matter much either way.